### PR TITLE
[integration][windows] -  Fixed parsing of winlog.event_data.MemberName in forwarded data stream when extra commas are present

### PR DESCRIPTION
**Pull request by ShourieG**
## Type of change
Please label this PR with one of the following labels, depending on the scope of your change:
- Bug

## Proposed commit message
ISSUE: The Windows integration did not parse winlog.event_data.MemberName properly and map it to user.target.name  when user CN contained extra "," characters. For example ```CN=Reyes\, Elena (FFF),OU=Basic,OU=Domain Users,DC=ddd,DC=ccc,DC=fff``` will be parsed to user.target.name ```Reyes\,``` which is wrong.

FIX:  Fixed parsing of winlog.event_data.MemberName in forwarded data stream by removing the old splitToken method and replaced it with a custom split processor logic that splits on the basis of commas but ignores escaped commas.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

-

### CODE REVIEW

File: packages/windows/data_stream/forwarded/elasticsearch/ingest_pipeline/security.yml [hunk 1]
```diff
@@ -2449,7 +2488,7 @@ processors:
           return;
         }
         if (ctx?.winlog?.event_data?.MemberName != null) {
-          def memberNameParts = ctx.winlog.event_data.MemberName.splitOnToken(",");
+          def memberNameParts = splitString(ctx.winlog.event_data.MemberName);
```
Comment (leehinman) [inline packages/windows/data_stream/forwarded/elasticsearch/ingest_pipeline/security.yml:2491]: I think we need to do the same thing in the System integration.  https://github.com/elastic/integrations/blob/a537327779334d94aeb6e91a15d43247dd290501/packages/system/data_stream/security/elasticsearch/ingest_pipeline/standard.yml#L2580

File: packages/windows/data_stream/forwarded/elasticsearch/ingest_pipeline/security.yml [hunk 2]
```diff
@@ -2441,6 +2441,45 @@ processors:
       tag: Copy MemberName to User and User to Group
       description: Copy MemberName to User and User to Group
       source: |-
+        def splitString(String input) {
+          // Count the number of parts
+          int count = 1;
+          for (int i = 0; i < input.length(); i++) {
+              if (Character.compare(input.charAt(i), ",".charAt(0))==0 && i < input.length() - 3 && Character.isUpperCase(input.charAt(i + 1))
```
Comment (leehinman) [inline packages/windows/data_stream/forwarded/elasticsearch/ingest_pipeline/security.yml:2448]: does it always have to be 2 Uppercase characters followed by Equals?  Does Windows ever do something like O=Internet,CN=Test User ?

**Review by leehinman — CHANGES_REQUESTED** [review]

**Review by efd6 — COMMENTED** [review]
It seems to me that this could be done with a split processor; something like `{"split": {"if": "ctx.winlog?.event_data?.MemberName != null", "field": "winlog.event_data.MemberName", "target_field": "_temp.MemberNameParts", "separator":"(?<!\\\\),"}` with `_temp.MemberNameParts` being used as the direct source of the script's `memberNameParts` var.

**Comment by ShourieG** [comment]
> Since the original issue shows that the comma is preceded by a backslash, it seems to me that this could be done with a split processor; something like `{"split": {"if": "ctx.winlog?.event_data?.MemberName != null", "field": "winlog.event_data.MemberName", "target_field": "_temp.MemberNameParts", "separator":"(?<!\\\\),"}` with `_temp.MemberNameParts` being used as the direct source of the script's `memberNameParts` var.

@efd6 true, but we don't know if that's always the case or not, hence more of a generic approach was taken.

File: packages/windows/data_stream/forwarded/elasticsearch/ingest_pipeline/security.yml (hunk 2 above)
Comment (ShourieG) [reply packages/windows/data_stream/forwarded/elasticsearch/ingest_pipeline/security.yml:2448]: @elasticmachine I'm not sure as my knowledge is limited with regards to this, @efd6 do you have any idea ?

File: packages/windows/data_stream/forwarded/elasticsearch/ingest_pipeline/security.yml (hunk 1 above)
Comment (ShourieG) [reply packages/windows/data_stream/forwarded/elasticsearch/ingest_pipeline/security.yml:2491]: I'll open a separate issue for the system integration as this PR is related to a specific SDH.

**Comment by efd6** [comment]
> @efd6 true, but we don't know if that's always the case or not, hence more of a generic approach was taken.

I have asked at the original issue. If it is guaranteed, the simpler approach would be preferable.

File: packages/windows/data_stream/forwarded/elasticsearch/ingest_pipeline/security.yml (hunk 1 above)
Comment (ShourieG) [reply packages/windows/data_stream/forwarded/elasticsearch/ingest_pipeline/security.yml:2491]: Opened this separate issue: https://github.com/elastic/integrations/issues/8419

**Comment by andrewkroh** [comment]
https://learn.microsoft.com/en-us/previous-versions/windows/desktop/ldap/distinguished-names

File: packages/windows/data_stream/forwarded/elasticsearch/ingest_pipeline/security.yml (hunk 2 above)
Comment (leehinman) [reply packages/windows/data_stream/forwarded/elasticsearch/ingest_pipeline/security.yml:2448]: Given the doc @andrewkroh linked to, I'm in favor of the approach @efd6 suggested.  It handles the ',' in the attribute value and allows for the attributes to be more than 2 uppercase characters.

**Comment by efd6** [comment]
The key is in the table [here](https://learn.microsoft.com/en-us/previous-versions/windows/desktop/ldap/distinguished-names). `,` is reserved, and so MUST be escaped. This means the split processor approach is valid.

**Comment by ShourieG** [comment]
@efd6, @leehinman I've updated the PR with the split processor logic as suggested.

**Review by ycombinator — APPROVED** [review]
LGTM.

**Review by leehinman — APPROVED** [review]
LGTM

**Review by kcreddy — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**